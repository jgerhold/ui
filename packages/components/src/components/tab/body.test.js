import { mount } from '@vue/test-utils'

import TabHeader from './Header'

describe('TabHeader', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        }
      })

      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    test(`should render with prop backgroundActive`, async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        },
        propsData: {
          backgroundActive: '#fff'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
