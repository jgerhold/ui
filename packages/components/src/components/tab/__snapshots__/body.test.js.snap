// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabHeader props should render with prop backgroundActive 1`] = `
<ul
  class="tab-header"
  role="tablist"
>
  <span
    class="header-shadow"
  />
   
  <span>
    Inner Tab Header
  </span>
</ul>
`;

exports[`TabHeader props should render without props 1`] = `
<ul
  class="tab-header"
  role="tablist"
>
  <span
    class="header-shadow"
  />
   
  <span>
    Inner Tab Header
  </span>
</ul>
`;
